course_catalog/tasks.py

```diff
@@ -28,6 +28,7 @@
 def get_mitx_data():
     """Task to sync mitx data with the database"""
     pipelines.mitx_etl()
+    mitx.dedupe_courses()
```

Comment:
I think deduplicating the courses at the end of everything could be problematic, a few issues I could see arising from this:

- Courses will reappear in search results for a few minutes until `mitx_etl()` returns and we start deduplicating
- Any failure along the way could cause this task to exit and leave us in a partial state (e.g. multiple courses showing up again)

I think we need to move the deduplication to an earlier stage

Reply:
I updated the pr to not update any of the duplicate courses in elasticsearch until the deduplication step happens. This prevents duplicate courses from appearing while the tasks runs. If the task fails  before the dedupe step, courses with duplicates will be not updated in the search but no duplicates will appear